warehouse/oidc/views.py

```diff
@@ -71,96 +96,106 @@ def oidc_audience(request):
     require_csrf=False,
     has_translations=True,
 )
-def mint_token_from_oidc(request):
-    def _invalid(errors):
-        request.response.status = 422
-        return {"message": "Token request failed", "errors": errors}
-
+def mint_token_from_oidc_github(request: Request):
```

Comment:
I think instead we should probably make a generic non-GitHub-specific endpoint that accepts any OIDC token in the body and attempts to mint a token for it based on properties of the token (specifically, the `iss` claim should allow us to pick the right publisher type)

I don't see a great reason to maintain a separate endpoint for every OIDC provider we want to support here, that was probably not necessary in the GitHub implementation.

(cc @woodruffw for any thoughts as well)

Reply:
Yes, agreed -- the main reason I did that originally was to avoid snooping the unverified JWT claims before verifying them, but that ship has sailed elsewhere in the implementation IIRC (and is not a security risk, so long as we perform proper key segregation and add both positive and negative tests).

We'll probably have to keep the GitHub-specific endpoint around for a while because of existing users (including some people who have reimplemented the token exchange flow), but it can be a thin shim over the "generic" endpoint :slightly_smiling_face: